aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -640,9 +640,18 @@ Tensor scaled_dot_product_attention(
   int64_t choice_int = static_cast<int64_t>(sdp::SDPBackend::math);
   if (query_.device().type() == DeviceType::CUDA
       || query_.device().type() == DeviceType::CPU
-      || query_.device().type() == DeviceType::HIP){
-    choice_int = _fused_sdp_choice_stub(query_.device().type(),
+      || query_.device().type() == DeviceType::HIP
+      || query_.device().type() == DeviceType::PrivateUse1){
+    if (query_.device().type() == DeviceType::PrivateUse1){
+      try {
+        choice_int = _fused_sdp_choice_stub(query_.device().type(),
```

Comment:
why do we need the try catch, is that because some backend will not implement and we  want them to fall through to the math backend?

Reply:
Yes, you are right

Reply:
@drisspg Looking forward to your reply.

Reply:
Can you abstract out this logic out to separate function, something like:

```C++
  auto device_type = query_.device().type();
  if (device_type == DeviceType::CUDA || device_type == DeviceType::CPU ||
      device_type == DeviceType::HIP ||
      device_type == DeviceType::PrivateUse1) {
    if (device_type == DeviceType::PrivateUse1) {
      choice_int = handle_private_use(
          query_, key, value, attn_mask_, dropout_p, is_causal, scale);
    } else {
    choice_int = _fused_sdp_choice_stub(device_type, query_, key, value, attn_mask_, dropout_p, is_causal, scale);
  }
 ```

Reply:
OK, I have already modified it @drisspg

Reply:
f you have any further questions, feel free to bring them up. @drisspg

Reply:
Hey sorry just wanted to see if ci had any complaints